addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -448,25 +450,51 @@ export const editorCommands = {
     },
     unlink: editor => {
         const sel = editor.document.getSelection();
-        // we need to remove the contentEditable isolation of links
-        // before we apply the unlink, otherwise the command is not performed
-        // because the content editable root is the link
-        const closestEl = closestElement(sel.focusNode, 'a');
-        if(closestEl) {
-            closestEl.removeAttribute('class'); // To prevent firefox from adding unnecessary <span>.
-        }
-        if (closestEl && closestEl.getAttribute('contenteditable') === 'true') {
-            editor._activateContenteditable();
+        const isCollapsed = !!sel.isCollapsed;
+        if (!isCollapsed) {
+            getDeepRange(editor.editable, { sel, splitText: true, select: true });
+            // Split leading and trailing links.
+            let { anchorNode, focusNode, anchorOffset, focusOffset } = sel;
+            const direction = getCursorDirection(anchorNode, anchorOffset, focusNode, focusOffset);
+            let startLink = closestElement(anchorNode, 'a');
+            if (startLink) {
+                startLink = splitAroundUntil(anchorNode, startLink);
+                const range = sel.getRangeAt(0);
+                anchorNode = startLink;
+                anchorOffset = direction === DIRECTIONS.RIGHT ? 0 : nodeSize(startLink);
+                range.setStart(anchorNode, anchorOffset);
+                sel.removeAllRanges();
+                sel.addRange(range);
+                sel.extend(focusNode, focusOffset);
+                getDeepRange(editor.editable, { sel, select: true });
+            }
+            let endLink = closestElement(focusNode, 'a');
+            if (endLink) {
+                endLink = splitAroundUntil(focusNode, endLink);
+                const range = sel.getRangeAt(0);
+                range.setStart(anchorNode, anchorOffset);
+                sel.removeAllRanges();
+                sel.addRange(range);
+                sel.extend(endLink, direction === DIRECTIONS.RIGHT ? nodeSize(endLink) : 0);
+            }
         }
-        if (sel.isCollapsed) {
+        getDeepRange(editor.editable, { sel, select: true });
+        const links = [...new Set((isCollapsed ? [sel.anchorNode] : getSelectedNodes(editor.editable))
+            .map(node => closestElement(node, 'a')).filter(a => a))];
+        if (links.length) {
+            // We need to remove the contentEditable isolation of links before
+            // we apply the unlink, otherwise the command is not performed
+            // because the content editable root is the link.
+            if (links[0].getAttribute('contenteditable') === 'true') {
+                editor.resetContenteditableLink();
+                editor._activateContenteditable();
+            }
```

Comment:
I tend to avoid writing `[0]` (and it won't work with a set anyway), I suggest:
```js
const [link] = link;
if (link && link.getAttribute(...
```
However, I'm not even sure this code still applies since we modify the DOM by hand rather than relying on the contentEditable `execCommand` mechanism.